Mathlib/Algebra/Module/Length.lean

```diff
@@ -0,0 +1,119 @@
+import Mathlib.Order.KrullDimension
+import Mathlib.Order.TrimmedLength
+import Mathlib.LinearAlgebra.Span.Basic
+import Mathlib.Data.ENat.Lattice
+import Mathlib.Algebra.Homology.ShortComplex.Basic
+import Mathlib.Order.Hom.Basic
+import Mathlib.Algebra.Category.ModuleCat.Basic
+import Mathlib.Algebra.Homology.ShortComplex.ModuleCat
+import Mathlib.Data.ENat.Lattice
+
+/-!
+# The length of a module
+
+-/
+
+variable {R : Type*} [Ring R] {M M' : Type*} [AddCommGroup M] [AddCommGroup M']
+variable [Module R M] [Module R M']
+
+open Order
+
+variable (R M) in
+/--
+The length of a module M is defined to be the supremum of lengths of chains of submodules of M. We
+define this using the existing krull dimension api, and as a result this takes values in
+WithBot ℕ∞ in spite of the fact that there is no module with length equal to ⊥.
```

Comment:
I think we should make `Module.length` take value in `ℕ∞` using `WithBot.unbot` (`unbotD` is also acceptable).